%YAML:1.0

#--------------------------------------------------------------------------------------------
# ROS topics
#--------------------------------------------------------------------------------------------
imuTopic: "/imu0"
leftImageTopic: "/cam0/image_raw" #"/camera/color/image_raw"

readFromRosBag: 1   # 1: read from rosbag; 0: subscribe from rosbag (you should 'rosbag play ...')
# note: 'rosBag' is configurated in launch file
#rosBag: "/home/hri/Documents/SLAM_Dataset/EuRoC/rosbag_file/V1_03_difficult.bag"

#--------------------------------------------------------------------------------------------
# Paramters!
#--------------------------------------------------------------------------------------------
downSampleRate: 1  # default: 1
KeyPointNumber: 500
ThresholdOfPredictNewKeyPoint: 1.0
HalfPatchSize: 5

estimateTD: 0           # whether to estimate time-delay. 0: not estimate; 1: estimate

# note: 'manuallyAddTimeDelay' is configurated in launch file
#manuallyAddTimeDelay: 0.060 # manually added time delay to image timestamps, unit: s

# note: 'output_file' is configurated in launch file
#output_file: "/output/"

#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------
Camera.type: "PinHole"

# Camera calibration and distortion parameters (OpenCV) 
Camera.fx: 458.654
Camera.fy: 457.296
Camera.cx: 367.215
Camera.cy: 248.375

Camera.k1: -0.28340811
Camera.k2: 0.07395907
Camera.p1: 0.00019359
Camera.p2: 1.76187114e-05

# Camera resolution
Camera.width: 752
Camera.height: 480

# Camera frames per second 
Camera.fps: 20

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 1

# Transformation from camera to body-frame (imu)
#Tbc: !!opencv-matrix
#   rows: 4
#   cols: 4
#   dt: f
#   data: [0.0148655429818, -0.999880929698, 0.00414029679422, -0.0216401454975,
#         0.999557249008, 0.0149672133247, 0.025715529948, -0.064676986768,
#        -0.0257744366974, 0.00375618835797, 0.999660727178, 0.00981073058949,
#         0.0, 0.0, 0.0, 1.0]

Tbc:
    [0.0148655429818, -0.999880929698, 0.00414029679422, -0.0216401454975,
     0.999557249008, 0.0149672133247, 0.025715529948, -0.064676986768,
    -0.0257744366974, 0.00375618835797, 0.999660727178, 0.00981073058949,
     0.0, 0.0, 0.0, 1.0]

# IMU noise
IMU.NoiseGyro: 1.7e-4       # rad/s/sqrt(Hz)
IMU.NoiseAcc: 2.0000e-3     # m/s^2/sqrt(Hz)
IMU.GyroWalk: 1.9393e-05    # rad/s^2/sqrt(Hz)
IMU.AccWalk: 3.0000e-03     # m/s^3/sqrt(Hz)
IMU.Frequency: 200

##--------------------------------------------------------------------------------------------
## ORB Parameters
##--------------------------------------------------------------------------------------------

## ORB Extractor: Number of features per image
#ORBextractor.nFeatures: 1000 # 1000

## ORB Extractor: Scale factor between levels in the scale pyramid
#ORBextractor.scaleFactor: 1.2

## ORB Extractor: Number of levels in the scale pyramid
#ORBextractor.nLevels: 8

## ORB Extractor: Fast threshold
## Image is divided in a grid. At each cell FAST are extracted imposing a minimum response.
## Firstly we impose iniThFAST. If no corners are detected we impose a lower value minThFAST
## You can lower these values if your images have low contrast
#ORBextractor.iniThFAST: 20
#ORBextractor.minThFAST: 7

##--------------------------------------------------------------------------------------------
## Viewer Parameters
##--------------------------------------------------------------------------------------------
#Viewer.KeyFrameSize: 0.05
#Viewer.KeyFrameLineWidth: 1
#Viewer.GraphLineWidth: 0.9
#Viewer.PointSize:  2
#Viewer.CameraSize: 0.08
#Viewer.CameraLineWidth: 3
#Viewer.ViewpointX: 0
#Viewer.ViewpointY: -0.7
#Viewer.ViewpointZ: -3.5 # -1.8
#Viewer.ViewpointF: 500

